### PR TITLE
Add Tpch Test for native Parquet reader

### PR DESCRIPTION
**Pull request by majetideepak**
Add Tpch Test for native Parquet reader.
Add use_native_parquet_reader flag to TpchBenchmark.

### CODE REVIEW

File: velox/common/base/SimdUtil-inl.h [hunk 1]
```diff
@@ -294,7 +294,8 @@ inline bool copyNextWord(void*& to, const void*& from, int32_t& bytes) {
 } // namespace detail
 
 template <typename A>
-void memcpy(void* to, const void* from, int32_t bytes, const A& arch) {
+FOLLY_ALWAYS_INLINE void
```
Comment (Yuhta) [inline velox/common/base/SimdUtil-inl.h:297]: Did you notice big difference for inlining this?  This function seems too large to benefit from inlining.

File: velox/common/base/tests/SimdUtilTest.cpp [hunk 2]
```diff
@@ -333,3 +333,7 @@ TEST_F(SimdUtilTest, Batch64_memory) {
 }
 
 } // namespace
+
+void cpyf(void* d, void* s, int c) {
+  facebook::velox::simd::memcpy(d, s, c);
```
Comment (Yuhta) [inline velox/common/base/tests/SimdUtilTest.cpp:338]: Where is this used?

File: velox/dwio/parquet/reader/ParquetReader.cpp [hunk 3]
```diff
@@ -506,13 +506,29 @@ ParquetRowReader::ParquetRowReader(
 void ParquetRowReader::filterRowGroups() {
   auto scanSpec = options_.getScanSpec();
   auto rowGroups = readerBase_->fileMetaData().row_groups;
+  int32_t rangeBegin = -1;
+  int32_t rangeEnd = -1;
+  for (auto i = 0; i < rowGroups_.size(); ++i) {
+    VELOX_CHECK(rowGroups_[i].__isset.file_offset);
+    auto fileOffset = rowGroups_[i].file_offset;
+    if (fileOffset >= options_.getOffset() &&
```
Comment (Yuhta) [inline velox/dwio/parquet/reader/ParquetReader.cpp:514]: We can merge this condition into line 526 below

**Review by Yuhta — COMMENTED** [review]
Looks good except a few questions, and missing null annotations that breaking the build on CircleCI need to be added.

**Comment by majetideepak** [comment]
CC: @pramodsatya

File: velox/dwio/parquet/reader/ParquetData.cpp [hunk 4]
```diff
@@ -85,8 +85,10 @@ void ParquetData::enqueueRowGroup(
   }
   VELOX_CHECK_GE(chunkReadOffset, 0);
 
-  uint64_t readSize = std::min(
-      metaData.total_compressed_size, metaData.total_uncompressed_size);
+  uint64_t readSize = (metaData.total_uncompressed_size > 0)
```
Comment (majetideepak) [inline velox/dwio/parquet/reader/ParquetData.cpp:88]: DuckDB does not set `total_uncompressed_size`. Hence this is required.

**Comment by majetideepak** [comment]
@Yuhta Can you take a look again? Thanks.

File: velox/dwio/parquet/reader/ParquetData.cpp (hunk 4 above)
Comment (Yuhta) [reply velox/dwio/parquet/reader/ParquetData.cpp:88]: `metaData.__isset.total_uncompressed_size`?

**Review by Yuhta — COMMENTED** [review]
Looks good except one tiny suggestion

**Comment by majetideepak** [comment]
> metaData.__isset.total_uncompressed_size?

`total_uncompressed_size` is a required field. So it will not have an `isset` field.
https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L737

**Pull request closed without merging**